modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/NamingComponent.java

```diff
@@ -17,6 +27,48 @@
  * @author Heiko W. Rupp
  */
 public class NamingComponent extends BaseComponent implements OperationFacet {
+    private static final String BINDING_TYPE = "Binding";
+    private static final String ENVIRONMENT = "environment";
+    @Override
+    public CreateResourceReport createResource(CreateResourceReport report) {
+        if (report.getResourceType().getName().equals(BINDING_TYPE)) {
+            Configuration resourceConfiguration = report.getResourceConfiguration();
+            PropertyList prop = (PropertyList)resourceConfiguration.get(ENVIRONMENT);
+            List<Property> embeddedProps = prop.getList();
+
+            /* Modify definition on-fly*/
+            ConfigurationDefinition configDef = report.getResourceType().getResourceConfigurationDefinition().copy();
+            Map<String, PropertyDefinition> definitions = configDef.getPropertyDefinitions();
+            PropertyDefinitionList envDefinition = (PropertyDefinitionList)definitions.get(ENVIRONMENT);
+            PropertyDefinitionMap envDefinitionMap = new PropertyDefinitionMap(envDefinition.getName(),
+                    envDefinition.getDescription(), envDefinition.isRequired());
+            definitions.put(ENVIRONMENT, envDefinitionMap);
+
+            /* Modify resource configuration on-fly*/
+            PropertyMap environmentMap = new PropertyMap();
+            environmentMap.setName(ENVIRONMENT);
+            Map<String, Property> results = new HashMap<String, Property>();
+            for (Property inner : embeddedProps) {
+                PropertyMap propMap = (PropertyMap) inner;
+                PropertySimple propName = ((PropertySimple) propMap.get("name"));
+                PropertySimple propValue = ((PropertySimple) propMap.get("value"));
+                PropertySimple propS = new PropertySimple();
+                propS.setName(propName.getStringValue());
```

Comment:
Are these null safe?

Reply:
Good point, I think I need to add null validations here to avoid NPE.